### PR TITLE
Formatting balooza

### LINKED ISSUE #45: Create Blog Component
This would be the `<section></section>` that we currently have associate with the blog route.

We can extract this into its own component and use one of the loops functions to loop over some details data and have only one component show all blogs.

From here, we can have an associated 'Blog Form' component that will allow someone to create a new blog. Then the new blog would get added to the .json data that gets populated. However, it would probably be better practice if we branch off here and add some backend connections:
1. With this as backend, we would have the Blog component pull and populate the blog data from the database.
2. Anytime a form is submitted, it can be added to the database and then showcased on the blog page.
  - Where should this live?
  - Do we want to add markdown support?

EDIT:
There are several different flavors of "blog" components that we have. We currently have a component that pulls all of the blog posts from our api. We need to add the following conditions / new formats:
  - The new line character is not working in the blog posts. This needs to be fixed (i.e., whenever there is `\n` there should be a new line).
  - Blog posts should be cut off after a certain number of characters and should also have some type of "read more" button that takes the the user to a page that is solely dedicated to that one blog post.
    - This page should also have a comments section (probably at the bottom).
      - This comments section should have an edit button that is displayed if and only if 
        - The comment was written by a non admin and that non admin is the author of the post, or
        - The user is an admin user.
  - Blog posts should have an edit button only visible to admin users.
  - Create a new page where users can write new blogs, add tags, blah blah blah.
    - Need an add blog button only visible to admin users
      - Have no idea where this would go. Maybe the navbar?

### PR DESCRIPTION
**Pull request by StephenODea54**
- Fix new line character formatting in blog posts
- Cut off blog posts after a certain number of characters
- Add `SingleBlogPost` component that shows only one blog post in its entirety
- Add comments to the single blog post page
- Add edit and delete buttons for posts and comments for admin users (or for users that are responsible for their own post)
- Make the `post` navbar link only visible to admin users

Was going to refactor bc it needs it bad, but will defer to later / someone else since we need to finish this up.

Closes #45 #58

### CODE REVIEW

File: client/src/Layout.jsx
Comment (jpatterson933) [inline client/src/Layout.jsx:1]: oh im really excited to be an admin lol

File: client/src/components/singleBlogPost/index.jsx
Comment (jpatterson933) [inline client/src/components/singleBlogPost/index.jsx:1]: drum rolllllll the moment we've all been waiting for!! Amazing work

**Review by jpatterson933 — APPROVED** [review]
Hopefully we can get together later today and see this live in action! Deadline is less than 24 hrs

File: client/src/components/singleBlogPost/index.jsx
Comment (StephenODea54) [reply client/src/components/singleBlogPost/index.jsx:1]: Thank you sir!

**Review by nathanjslim — APPROVED** [review]

**Pull request merged**